module/zfs/dnode.c

```diff
@@ -1815,6 +1880,8 @@ dnode_free_range(dnode_t *dn, uint64_t off, uint64_t len, dmu_tx_t *tx)
 		if (last != first)
 			dnode_dirty_l1(dn, last, tx);
 
+		dnode_dirty_l1range(dn, first, last, tx);
+
```

Comment:
Question for @pcd1193182 : what does the loop on line 1886 do that dnode_dirty_l1() and dnode_dirty_l1range() invocations just above did not accomplish ? It seems that the look below still deals with blocks from the same range ?